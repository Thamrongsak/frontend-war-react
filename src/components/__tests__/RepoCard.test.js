import React from 'react'

import RepoCard from '../RepoCard.react'
import ReactTestRenderer from 'react-test-renderer'

import { MemoryRouter } from 'react-router-dom'

describe('RepoCard', () => {
    it('should render empty repo object', () => {
        const tree = ReactTestRenderer.create(
            <MemoryRouter>
                <RepoCard repository={ {} } />
            </MemoryRouter>
        )
        expect(tree).toMatchSnapshot()
    })

    it('should render correctly', () => {
        const repo = {
            name: "React",
            description: "Lib for create view",
            stargazers_count: 60000,
            open_issues_count: 400,
            forks_count: 1200,
            pull_requests_count: 200
        }
        const tree = ReactTestRenderer.create(
            <MemoryRouter>
                <RepoCard repository={ repo } />
            </MemoryRouter>
        )
        expect(tree).toMatchSnapshot()
    })
})