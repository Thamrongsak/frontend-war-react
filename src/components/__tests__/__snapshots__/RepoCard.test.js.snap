exports[`RepoCard should render correctly 1`] = `
<div
  className="col-6">
  <div
    className="card">
    <h3
      className="card-header">
      React
    </h3>
    <div
      className="card-block">
      <p
        className="card-text">
        Lib for create view
      </p>
      <p
        className="card-text">
        Stars: 
        60000
      </p>
      <p
        className="card-text">
        Open Issues: 
        400
      </p>
      <p
        className="card-text">
        Forks: 
        1200
      </p>
      <p
        className="card-text">
        Pull Requests: 
        200
      </p>
    </div>
    <div
      className="card-footer text-muted">
      <a
        className="btn btn-primary"
        href="/stargazers/undefined"
        onClick={[Function]}>
        View enemies
      </a>
    </div>
  </div>
</div>
`;

exports[`RepoCard should render empty repo object 1`] = `
<div
  className="col-6">
  <div
    className="card">
    <h3
      className="card-header">
      -
    </h3>
    <div
      className="card-block">
      <p
        className="card-text">
        -
      </p>
      <p
        className="card-text">
        Stars: 
        0
      </p>
      <p
        className="card-text">
        Open Issues: 
        0
      </p>
      <p
        className="card-text">
        Forks: 
        0
      </p>
      <p
        className="card-text">
        Pull Requests: 
        0
      </p>
    </div>
    <div
      className="card-footer text-muted">
      <a
        className="btn btn-primary"
        href="/stargazers/undefined"
        onClick={[Function]}>
        View enemies
      </a>
    </div>
  </div>
</div>
`;
